### ISSUE TITLE
[BUG] tribe29.checkmk.host is not idempotent

### ISSUE THREAD

**Issue by xenorites** [tribe29/ansible-collection-tribe29.checkmk#58]
**Describe the bug**
When running the tribe29.checkmk.host for a host that already exists in check_mk (with `state: present`) the module returns an error.

**Component Name**
`tribe29.checkmk.host`

**Ansible Version**
```
ansible [core 2.12.5]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/xenorites/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.10/site-packages/ansible
  ansible collection location = /home/xenorites/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.4 (main, Mar 25 2022, 00:00:00) [GCC 12.0.1 20220308 (Red Hat 12.0.1-0)]
  jinja version = 3.0.3
  libyaml = True
```

**Checkmk Version**

```
2.0.0p25 (CRE)
```

**Collection Version**
```
0.3.3
```

**Expected behavior**
Ansible should return an "OK" status for idempotency. 

**Actual behavior**
Ansible returns an error:
```
FAILED! => {"changed": false, "msg": "Error calling API. HTTP code 400. Details: b'{\"title\": \"Invalid move action\", \"status\": 400, \"detail\": \"The host is already part of the specified target folder\"}', "}
```

**Comment by robin-tribe29** [comment]
Probably a duplicate of #41.
Please test this with our 'devel' branch.
A release of the collection will happen in approx. two weeks.

**Comment by robin-tribe29** [comment]
HI @xenorites and thanks for your patience. What you did should work, but we released [v0.4.0](https://github.com/tribe29/ansible-collection-tribe29.checkmk/releases/tag/v0.4.0) yesterday, so you might want to check that one. If the issue then still persists, I will hand this over to a developer to take a look.

**Comment by xenorites** [comment]
Confirmed, the error does remain the same at this point.

**Comment by robin-tribe29** [comment]
Okay, thank you for testing again. I just received word, that some related bugs in Checkmk have been fixed, and the fixes will be present in Checkmk 2.1.0p5. This release is expected within two weeks, so I want to ask you again to check against that release once it is out. Thanks!

**Comment by cmasopust** [comment]
could not confirm that this bug is solved :(     

I'm using version 0.5.0 and it is still not idempotent, this time because of the leading slash...

**Comment by diademiemi** [comment]
As a temporary solution you can add
`failed_when: checkmk_agent_create_result.failed is true and "The host is already part of the specified target folder" not in checkmk_agent_create_result.msg` to the task. This will make it not fail if the host is already created.

@robin-tribe29 The agent role has this line in main.yml, if a fix is implemented in the module itself this line will be redundant and should also be removed.

**Comment by lgetwan** [comment]
Fixed with #147
